src/probnum/problem_zoo/linalg/_random_spd_matrix.py

```diff
@@ -0,0 +1,158 @@
+
+    # Draw orthogonal matrix with respect to the Haar measure
+    orth_mat = scipy.stats.special_ortho_group.rvs(dim, random_state=random_state)
+    spd_mat = orth_mat @ np.diag(spectrum) @ orth_mat.T
+
+    # Symmetrize to avoid numerically not symmetric matrix
+    # Since A commutes with itself (AA' = A'A = AA) the eigenvalues do not change.
+    return 0.5 * (spd_mat + spd_mat.T)
+
+
+def random_sparse_spd_matrix(
+    dim: IntArgType,
+    density: float,
+    chol_entry_min: float = 0.1,
+    chol_entry_max: float = 1.0,
+    random_state: Optional[RandomStateArgType] = None,
+) -> np.ndarray:
+    """Random sparse symmetric positive definite matrix.
+
+    Constructs a random sparse symmetric positive definite matrix for a given degree
+    of sparsity. The matrix is constructed from its Cholesky factor :math:`L`. Its
+    diagonal is set to one and all other entries of the lower triangle are sampled
+    from a uniform distribution with bounds :code:`[chol_entry_min, chol_entry_max]`.
+    The resulting sparse matrix is then given by :math:`A=LL^\\top`.
+
+    Parameters
+    ----------
+    dim
+        Matrix dimension.
+    density
+        Degree of sparsity of the off-diagonal entries of the Cholesky factor.
+        Between 0 and 1 where 1 represents a dense matrix.
+    chol_entry_min
+        Lower bound on the entries of the Cholesky factor.
+    chol_entry_max
+        Upper bound on the entries of the Cholesky factor.
+    random_state
+        Random state of the random variable. If None (or np.random), the global
+        :mod:`numpy.random` state is used. If integer, it is used to seed the local
+        :class:`~numpy.random.RandomState` instance.
+
+    See Also
+    --------
+    random_spd_matrix : Generate a random symmetric positive definite matrix.
+
+    Examples
+    --------
+    >>> from probnum.problem_zoo.linalg import random_sparse_spd_matrix
+    >>> sparsemat = random_sparse_spd_matrix(dim=5, density=0.1, random_state=42)
+    >>> sparsemat
+    array([[1.        , 0.        , 0.        , 0.        , 0.        ],
+           [0.        , 1.        , 0.        , 0.        , 0.        ],
+           [0.        , 0.        , 1.        , 0.        , 0.24039507],
+           [0.        , 0.        , 0.        , 1.        , 0.        ],
+           [0.        , 0.        , 0.24039507, 0.        , 1.05778979]])
+    """
+
+    # Initialization
+    random_state = _utils.as_random_state(random_state)
+    if not 0 <= density <= 1:
+        raise ValueError(f"Density must be between 0 and 1, but is {density}.")
+    chol = np.eye(dim)
+    num_off_diag_cholesky = int(0.5 * dim * (dim - 1))
+    num_nonzero_entries = int(num_off_diag_cholesky * density)
+
+    if num_nonzero_entries > 0:
+        # Draw entries of lower triangle (below diagonal) according to sparsity level
+        entry_ids = np.mask_indices(n=dim, mask_func=np.tril, k=-1)
+        _idx = random_state.choice(
```

Comment:
please dont hate me for this comment, but I dont think you need the leading underscore? Seems to me like the namespace of this variable does not clash with anything.